peekingduck/pipeline/nodes/output/screen.py

```diff
@@ -49,15 +51,33 @@ def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
         super().__init__(config, node_path=__name__, **kwargs)
         cv2.namedWindow(self.window_name, cv2.WINDOW_NORMAL)
         cv2.moveWindow(self.window_name, self.window_loc["x"], self.window_loc["y"])
+        self.aspect_ratio_thres = 5e-3
+        self.current_filename = None
 
     def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
         """Show the outputs on your display"""
-
         img = inputs["img"]
-        if self.window_size["do_resizing"]:
-            img = cv2.resize(
-                img, (self.window_size["width"], self.window_size["height"])
+        if not self.window_size["do_resizing"]:
+            img_height, img_width, _ = img.shape
+            if inputs["filename"] is not self.current_filename:
+                # Initialize the window size for every new video
+                cv2.resizeWindow(self.window_name, img_width, img_height)
+                self.current_filename = inputs["filename"]
+            else:
+                # Check the current window size, resize it if aspect ratio is not the same as image
+                aspect_ratio = img_width / img_height
+                _, _, win_width, win_height = cv2.getWindowImageRect(self.window_name)
+                if abs(win_width / win_height - aspect_ratio) > self.aspect_ratio_thres:
+                    win_height = int(win_width // aspect_ratio)
+                    cv2.resizeWindow(self.window_name, win_width, win_height)
```

Comment:
Assume a 400 x 400 (W x H) video is playing, if I were to resize the window to a short rectangle (maybe 400 x 50), the window will then be automatically resized to a tiny square (I assume 50 x 50). Is this expected behavior?

Reply:
Currently, the window will only be resized if there is a change in width, when `do_resizing` is set to `False`